src/identity/configuration/KeyGeneratingIdpConfigurationGenerator.ts

```diff
@@ -34,16 +37,16 @@ export class KeyGeneratingIdpConfigurationGenerator extends IdpConfigurationGene
   }
 
   // There is a typing difficulty with JSONWebKeySet, thus the "any"
-  private async generateJwks(): Promise<any> {
+  private async generateJwks(): Promise<{ keys: JWK[] }> {
     // Check to see if the keys are already saved
     const jwks = await this.storage.get(this.getJwksKey());
     if (jwks) {
       return jwks;
     }
     // If they are not, generate and save them
-    const keystore = JWK.createKeyStore();
-    await keystore.generate('RSA');
-    const newJwks = keystore.toJSON(true);
+    const { privateKey } = await generateKeyPair('RS256');
+    const jwk = await fromKeyLike(privateKey);
+    const newJwks = { keys: [ jwk ]};
     await this.storage.set(this.getJwksKey(), newJwks);
     return newJwks;
```

Comment:
```suggestion
    await this.storage.set(this.getJwksKey(), { keys: [ jwk ]});
    return this.storage.get(this.getJwksKey());
```

Maybe?

Reply:
Since the storage might be accessing a ResourceStore this could be more costly.

Reply:
Yeah, I'd advocate for consistency and readability of code over small performance optimisation...

Reply:
I actually find the first one more readable since it's clear what we put in the variable. 😄 This is also similar to how the implementation was originally.

Reply:
Yeah, I always try to get rid of as many variables as possible, since it feels like the wrong kind of cognitive load. On second read, I'd probably even get rid of `jwk` and replace it directly with the `await fromKeyLike(privateKey)`.

But I admit it's mostly preference based ^^,

Don't hesitate to resolve and merge (without commiting my suggestion)!